packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -541,13 +541,7 @@ export class StateMachine extends StateMachineBase {
       tracingConfiguration: props.tracingEnabled ? this.buildTracingConfiguration() : undefined,
       ...definitionBody.bind(this, this.role, props, graph),
       definitionSubstitutions: props.definitionSubstitutions,
-      encryptionConfiguration: props.kmsKey? {
-        kmsKeyId: props.kmsKey.keyArn,
-        kmsDataKeyReusePeriodSeconds: props.kmsDataKeyReusePeriodSeconds? props.kmsDataKeyReusePeriodSeconds.toSeconds() : this.defaultPeriodSeconds,
-        type: 'CUSTOMER_MANAGED_KMS_KEY',
-      }: {
-        type: 'AWS_OWNED_KEY',
-      },
+      encryptionConfiguration: constructEncryptionConfiguration(props, this.defaultPeriodSeconds),
```

Comment:
Sorry @wong-a, I saw your comment but pushed a commit addressing it without responding to the comment itself . I went ahead and created a method (`constructEncryptionConfiguration`) to create encryptionConfiguration since we need to do so for both Activity and StateMachine. I think it reads better, and this won't be breaking the current pattern (unlike previous commits) when using ternary operators.

Reply:
How does this work for Activity? `constructEncryptionConfiguration`'s first parameter accepts a `StateMachineProps` type

Reply:
Oh good catch! I'll update the method to accept both StateMachineProps and ActivityProps type

`export function constructEncryptionConfiguration(props: StateMachineProps | ActivityProps, defaultPeriodSeconds: number) {`